src-java/floodlightrouter-topology/floodlightrouter-storm-topology/src/main/java/org/openkilda/wfm/topology/floodlightrouter/FloodlightRouterTopology.java

```diff
@@ -41,500 +48,290 @@
-        List<String> kildaSwitchManagerTopics = new ArrayList<>();
-        for (String region: regions) {
-            kildaSwitchManagerTopics.add(
-                    Stream.formatWithRegion(topicsConfig.getTopoSwitchManagerRegionTopic(), region));
-        }
-        createKildaSwitchManagerReplyStream(builder, parallelism,
-                newParallelism, topicsConfig, kildaSwitchManagerTopics);
-
-        // Floodlight -- kilda.northbound --> Router
-        List<String> kildaNorthboundTopics = new ArrayList<>();
-        for (String region: regions) {
-            kildaNorthboundTopics.add(Stream.formatWithRegion(topicsConfig.getNorthboundRegionTopic(), region));
-        }
-        createKildaNorthboundReplyStream(builder, parallelism,
-                newParallelism, topicsConfig, kildaNorthboundTopics);
-
-        // Floodlight -- kilda.topo.nb --> Router
-        List<String> kildaNbWorkerTopics = new ArrayList<>();
-        for (String region: regions) {
-            kildaNbWorkerTopics.add(Stream.formatWithRegion(topicsConfig.getTopoNbRegionTopic(), region));
-        }
-        createKildaNbWorkerReplyStream(builder, parallelism, newParallelism, topicsConfig, kildaNbWorkerTopics);
-
-        // Part3 Request to Floodlights
-        // Storm -- kilda.speaker.flow --> Floodlight
-        createSpeakerFlowRequestStream(builder, newParallelism, topicsConfig);
-
-        // Storm -- kilda.speaker.flow.ping --> Floodlight
-        createSpeakerFlowPingRequestStream(builder, newParallelism, topicsConfig);
-
-        // Storm -- kilda.speaker --> Floodlight
-        createSpeakerRequestStream(builder, newParallelism, topicsConfig);
-
-        // Storm -- kilda.speaker.disco --> Floodlight
-        createSpeakerDiscoRequestStream(builder, newParallelism, topicsConfig);
-
-        // Storm <-- kilda.topo.disco -- Floodlight
-        List<String> kildaTopoDiscoTopics = new ArrayList<>();
-        for (String region : topologyConfig.getFloodlightRegions()) {
-            kildaTopoDiscoTopics.add(Stream.formatWithRegion(topicsConfig.getTopoDiscoRegionTopic(), region));
-
-        }
-        createKildaTopoDiscoReplyStream(builder, parallelism, newParallelism, topicsConfig, kildaTopoDiscoTopics);
-
-        createDiscoveryPipelines(builder, parallelism, topicsConfig, kildaTopoDiscoTopics);
-
-        // Storm -- kilda.stats.stats-request.priv --> Floodlight
-        createStatsStatsRequestStream(builder, parallelism, topicsConfig);
-
-        // Storm <-- kilda.fl-stats.switches.priv -- Floodlight
-        List<String> kildaFlStatsSwitchesTopics = regions.stream()
-                .map(region -> Stream.formatWithRegion(topicsConfig.getFlStatsSwitchesPrivRegionTopic(), region))
-                .collect(Collectors.toList());
-        createFlStatsSwitchesStream(builder, parallelism, topicsConfig, kildaFlStatsSwitchesTopics);
-
-        return builder.createTopology();
+    @Value
+    private static class TopologyOutput {
```

Comment:
It removes "knowledge" regarding specific bolts chain requirements from `createTopology` level and moves it on the level of the specific chain.

Yes, it can be relatively easily removed, but I would like to keep it in the current shape. If you have strict objections, I will remove it.